File: Business/Logic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Business
{
    public class Logic
    {
        public bool iscombinational(string s)
        {
            //add all the combinational gates
            if (s == "LUT4_L" || s == "LUT5" || s == "LUT5_L" || s == "LUT6" || s == "LUT6_L" || s == "INV" || s == "LUT3" || s == "LUT2" || s == "LUT4" || s == "MUXF5" || s == "MUXF8" || s == "MUXF7" || s == "LUT3_L" || s == "BUFGP")
                return true;
            else return false;
        }

        public bool issequential(string s)
        {
            //add all FFs
            if ((s == "FDD" || s == "FDE" || s == "FDC_1" || s == "FDC" || s == "FDCE" || s == "FD" || s == "FDPE" || s == "FDP" || s == "FDRE" || s == "FDRSE" || s == "FDR" || s == "FDS" || s == "FDSE" || s == "FDRS"))
                return true;
            else return false;
        }
    }
}

File: modnet/logic.py
class Logic(object):
    combinational = {
        "LUT4_L",
        "LUT5",
        "LUT5_L",
        "LUT6",
        "LUT6_L",
        "INV",
        "LUT3",
        "LUT2",
        "LUT4",
        "MUXF5",
        "MUXF8",
        "MUXF7",
        "LUT3_L",
        "BUFGP",
    }

    sequential = {
        "FDD",
        "FDE",
        "FDC_1",
        "FDC",
        "FDCE",
        "FD",
        "FDPE",
        "FDP",
        "FDRE",
        "FDRSE",
        "FDR",
        "FDS",
        "FDSE",
        "FDRS",
    }

    @classmethod
    def get_name_component(cls, line):
        return line.split(' ')[2]

    @classmethod
    def is_combinational(cls, line):
        name = cls.get_name_component(line)
        return name in cls.combinational

    @classmethod
    def is_sequential(cls, line):
        name = cls.get_name_component(line)
        return name in cls.sequential
